walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,84 @@
 // limitations under the License.
 
 package com.google.sps;
-
+import java.io.*; 
+import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Arrays;
+
 
 public final class FindMeetingQuery {
+
+  /*
+  * Receive events and a request and returns all the available time intervals
+  */  
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+      Collection<TimeRange> result;
+      Collection<String> requestAttendees = request.getAttendees();
+      Collection<String> requestOptionalAttendees = request.getOptionalAttendees();
+      
+      ArrayList<TimeRange> resultList = new ArrayList<TimeRange>();
+      ArrayList<TimeRange> resultWithOptional = new ArrayList<TimeRange>();
+      
+      int previousEventEndTime = TimeRange.START_OF_DAY;
+      int previousEventTimeOptionalAttendees = TimeRange.START_OF_DAY;
+      long requestedTime = request.getDuration();
+
+      result = Arrays.asList();
```

Comment:
Remove this line and initialize result with an empty list in line 33.

Reply:
Solved